src/Data/Integer/Properties.agda

```diff
@@ -1631,6 +1638,15 @@ i*j≡0⇒i≡0∨j≡0 i p with ℕ.m*n≡0⇒m≡0∨n≡0 ∣ i ∣ (abs-cong
 ... | inj₁ ∣i∣≡0 = inj₁ (∣i∣≡0⇒i≡0 ∣i∣≡0)
 ... | inj₂ ∣j∣≡0 = inj₂ (∣i∣≡0⇒i≡0 ∣j∣≡0)
 
+i*j≢0 : ∀ i j .{{_ : NonZero i}} .{{_ : NonZero j}} → NonZero (i * j)
+i*j≢0 i j rewrite abs-* i j = ℕ.m*n≢0 ∣ i ∣ ∣ j ∣
+
+i*j≢0⇒i≢0 : ∀ i {j} → .{{NonZero (i * j)}} → NonZero i
```

Comment:
I am doubtful whether this and the lemma below will actually work for instance search, because `_*_` is not a constructor and therefore Agda won't be able to infer what `i` and `j` are...

Do we have any uses for these? Both on concrete and abstract variables?

I think one solution might be make `j` explicit, but testing is required.

The easy solution is just not to include them as they're not actually used currently right?

Reply:
Lots of interesting commentary here.

Firstly, I introduced the three of them by analogy with the corresponding lemmas for `Nat`, to which they obviously delegate, and those haven't (so far!) been subject to the same criticism, and which have been quite heavily used (I think? 21 times after some mindless `grep`ping). Maybe need to revisit those?

Secondly, the parametrisation is identical to that in `Data.Nat.Properties`, and while I agree, none of these left-biased operations can infer their first argument (`Nat._+_`, `Nat._*_`, `List._++_`, etc.), Agda (usually) can infer the second given the first... maybe `Integer` is different? But since the definition of `_*_` itself delegates to both `Sign` and `Nat` multiplication, I had been supposing, *mutatis mutandis*...

Thirdly, the new lemma `sign-*` *does* have an irrelevant, instance) argument of the form `.{{NonZero (i * j)}}` so I suppose I had been conjecturing that whenever it gets used, there will typically be the need either to discharge it by delegation to `.{{NonZero i}}` and `.{{NonZero j}}`, or else to destruct it to get at those pieces (motivated by your criticism of #2230 I hadn't wanted to commit the user to any particular *a priori* analysis of how `.{{NonZero (i * j)}}` might arise...)

Finally, you're correct that no uses (yet!?), whereas there is the one use of `i*j≢0` in `Data.Integer.Divisibility.Signed`.

Conclusion: happy to remove, but with some nagging feeling that we might want them one day? In any case, YAGNI, for now at least. Will remove.